tmt/steps/execute/__init__.py

```diff
@@ -136,10 +136,12 @@ def go(self):
             raise tmt.utils.ExecuteError("No guests available for execution.")
 
         # Execute the tests, store results
-        for plugin in self.plugins():
-            plugin.go()
-            if isinstance(plugin, ExecutePlugin):
-                self._results = plugin.results()
+        for guest in self.plan.provision.guests():
+            for plugin in self.plugins():
```

Comment:
Should the `guest` loop be inside the `plugin` loop instead? Seems to me a bit more natural to go through individual `execute` plugins/configs and for each cover all relevant guests. This would also allow doing steps which need to go one after another in given order (e.g. client fetches a web page and then we check access log on the server to ensure the page was correctly served).

Reply:
Hm, I guess that both of the approaches make sense depending on how you think about it, I went with doing all steps on a single guest sequentially since this is what `finish` and `prepare` does. Since we do not do anything in parallel, from the point of connecting to the machines and having user-friendly logs, taking care of everything on a single guest at once seems a bit better to me for now.

Anyway, I will update the `multihost/web` test for now

Reply:
I see. I was writing the comment when I still thought we support multiple `execute` configs. This would allow to do things like:

* run client script on client
* run server script on server

But this is not the case. However, once we support multiple configs, this approach would be useful (and would work even before we have the support for parallelization).